Dockerfile.deps

```
@@ -0,0 +1,24 @@
+# Copyright 2021 Tetrate
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM gcr.io/cloud-builders/docker
```

maybe comment on the filename here? Is `Dockerfile.deps` a convention? Not trying to nitpick, just wondering if it is something cloudsmith related. The few places I've seen this convention was when there was already a `Dockerfile`

Maybe if we are interested in highlighting the target "deps" we can do this? and then name it normal dockerfile, if this isn't something specific..

```suggestion
FROM gcr.io/cloud-builders/docker AS deps
```

---

> I made some starting comments, but before reviewing a lot of python...
> 
> Wondering if there's a specific reason we are not using github actions for any steps here? It feels nice to have a way out of maintaining python scripts, even if only a few of them..
> 
> https://help.cloudsmith.io/docs/integrating-with-github-actions
> 
> It could probably help to have someone from cloudsmith review this PR to make sure we're doing least, but no less.

Honestly, I don't see any reason whatsoever to have these scripts, they only exist because the "build system" is Makefiles.

FWIW, the cloudsmith GHA/CircleCI integrations are a thin wrapper for the CLI (which is exactly what this bunch of ugly tech debt Python is doing), so it won't really make any logic simpler in the convoluted build/test/publish process.
Want me to cut a "refactor CI  / address the tech debt" issue?

---

yeah lets do it that way, any ideas you have in general into that issue would be great